### PR TITLE
UCT/IB/UD: Add missed UCT_CHECK_LENGTH for PUT Short

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Add missed UCT_CHECK_LENGTH for PUT Short

## Why ?

Fix it for consistency with other operations

## How ?

Call `UCT_CHECK_LENGTH` and expect that `sizeof(*neth) + sizeof(*put_hdr) + length` <= `iface->super.config.max_inline`

### CODE REVIEW

File: src/uct/ib/ud/accel/ud_mlx5.c [hunk 1]
```diff
@@ -336,6 +336,9 @@ uct_ud_mlx5_ep_put_short(uct_ep_h tl_ep, const void *buffer, unsigned length,
     uct_ud_send_skb_t *skb;
     size_t wqe_size;
 
+    UCT_CHECK_LENGTH(sizeof(*neth) + sizeof(*put_hdr) + length,
+                    0, iface->super.config.max_inline, "put_short");
```
Comment (hoopoepg) [inline src/uct/ib/ud/accel/ud_mlx5.c]: minor: alignment

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12775/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>ABORTED</b> on 1 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/fcaee5dcacc285b2f961e252729e05e9/raw/3156dd83c7f33b905ad2b8dfc1627616866a9334/_main) | :question: ABORTED

File: src/uct/ib/ud/accel/ud_mlx5.c [hunk 2]
```diff
@@ -336,6 +336,9 @@ uct_ud_mlx5_ep_put_short(uct_ep_h tl_ep, const void *buffer, unsigned length,
     uct_ud_send_skb_t *skb;
     size_t wqe_size;
 
+    UCT_CHECK_LENGTH(sizeof(*neth) + sizeof(*put_hdr) + length,
```
Comment (brminich) [inline src/uct/ib/ud/accel/ud_mlx5.c]: wrong check for put on mlx5
it is checked below on line 369

Comment (dmitrygx) [reply src/uct/ib/ud/accel/ud_mlx5.c]: we have the same checks for AM Short

Comment (brminich) [reply src/uct/ib/ud/accel/ud_mlx5.c]: ok, it is not real RDMA, then AM check should work.
anyway it seems excessive - this does the same as check on line 369

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12776/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/84bc6931e30acdfa2b1e46f9576cca528d11b492/_main) | :question: ABORTED
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/d3589962458356d38524263eba0f855f1d1b7e88/hpc-arm-cavium-jenkins_W1) | :question: ABORTED
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/48ffb4fd1de5cb52c4b7161193b44f43cc52e154/hpc-arm-cavium-jenkins_W3) | :question: ABORTED
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/f8d1f1f61927ec2926cf4bc3dfd600e5f527d5a8/hpc-arm-hwi-jenkins_W2) | :question: ABORTED
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/41d68718452665cf7b2f085d01391747de55b569/hpc-test-node-new_W0) | :question: ABORTED
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/276dab12c82e6eaf1a4deabe17aeead7eb3eb68c/hpc-test-node-new_W2) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/cf35eb94b3e31ff671cf49d07f10548e6d8c8998/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/1c8cdb96fffa6c325aa7d21ba22342cebb0d9892/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/45192fde89676fcd9bcdc92ae4d82732ad234249/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/9b9853b5f74f2476b9a857f89528f2da74344eae/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/c0e379be66dd59f122f6b05c7848cd64b2df185e/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/a3e6b4c4902ca79662964a2591af6287741885a6/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/a5c8ea261b6ea9b0c93b53240e1521868b03427e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/c0adac640c1c495442eeff5d90df752cbb3a2532/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/521a5d3359e579909cf0b18cbbc1549408fe0df7/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/e5115f8214ca5c88d5ffb9966731122220cc2d85/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/c46490e57c2a07a3f727f6bda0427dc83e3023f2/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/212098a79e1dbdebd3c90b670fb57f20b28cb93e/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/77b94c894d12d1fcb4e7b22ce70e076134d95493/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/9b84b4277563f1645810af53bd1a7d7978214744/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/fbcf31f0060a0ee445b758f26fb2660fe0104f73/hpc-arm-hwi-jenkins_W0) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/52c5ac73823ba30ddfb1b4a7e5ae7c3025df48f2/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/b923fc72ce54534b009207f36b9dc65b55b14455/hpc-arm-hwi-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/9c908f8b18a6c167f4d91ebb93084e9f88b51136/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/3cb10be10f49d71f464bd3faec79ff8bc2d6455b/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/a4ec4a8800333a131f24214a30360d74a05ed6f1/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/312f674a43bc0e7117d25c9ce5f64c9cf90aff83/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/828723dcb2aa8ba5e10415105a72d820e225e4de/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/f4d01a4773c70fd31b2780b69f2504d3/raw/a207f7c472a301def5b18fda914823fb54bece0b/hpc-test-node-new_W3) | :question: UNKNOWN

File: src/uct/ib/ud/accel/ud_mlx5.c [hunk 3]
```diff
@@ -213,7 +213,6 @@ uct_ud_mlx5_ep_am_short(uct_ep_h tl_ep, uint8_t id, uint64_t hdr,
     uct_ib_mlx5_inline_copy(am + 1, buffer, length, &iface->tx.wq);
 
     wqe_size += ctrl_av_size + sizeof(*inl);
-    UCT_CHECK_LENGTH(wqe_size, 0, UCT_IB_MLX5_MAX_SEND_WQE_SIZE, "am_short");
```
Comment (yosefe) [inline src/uct/ib/ud/accel/ud_mlx5.c]: why is this removed?

Comment (brminich) [reply src/uct/ib/ud/accel/ud_mlx5.c]: check on line 189 does the same

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12781/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/9e3c82eb5feadbd5768798c9eabda0cd8c28612a/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/787d82e9df5c974eb080ca6233e29e933d5f163c/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/914e9ae0b69460bf7464260cff9a065390f3f47a/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/d79c3160cb905b4933d99177ed3890e3cbbf62bd/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/b5dcc96520d8fcac45050a83747e888a9209041e/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/65e6994ac80156529baee4a2176fed60f4db8a02/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/fe86effb12c80bab03454a7b0910b5c311dd6d36/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/24fb387c2359b65e22eea64114dfe006332c43f9/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/ad05960ecca8864be65b07ced3c91d30ef87557b/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/b3ab6011c828f2ee5a18b5cd5e78bd3af4b28f55/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/3ba9b615428c5efae0c4084e48c6c8c9404dab44/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/b1169e68de9f1bbeed218caba9bda31d36e485ba/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/59efeb78a589f232288962adcd03dedc0adb0a06/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/4eaf8a8a177429fcdb3e4e3be321ca581f2709dc/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/468b13780eae89e1be8cad7d23a463999f16b010/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/a8bf4233ae177df868b7007bfc89e5a1a8201bad/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/6f95b104178b7d2389fcd8670f48a5e648d96601/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/fb15be494e2276a3c13d5de1dde44ece0a72affc/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/5b8e7bd91ffec70c6960954cd6304c4cee7c78f6/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/3dddef8fbdc9ac04c1b20b355e17936f86b1fc41/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/5abd5cc4b66e4ebe3c7bc57bd309b7a7a25c50f3/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/6fbd55535b028ac95814967fa7458fd6a8e0ba4f/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/451c5bb62cb70f17a2fe0445eb144852dad5b950/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/b56282234b09a9728add0a4b1b19d905d13844ae/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/c098dc0bcc180e61837dc09fe401e4c9b2a463af/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/b4a7e2b6b04f70b13020ebb249b31308a97c7c82/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/c34099264211d1959021a99f9016306e3824be89/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/2bd245979add12941e0849ee09e1d63cf3ecacd4/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/40a63c985905bcff0f6b04db9d250da3/raw/fc04bdd42641052bcc73f51061864c6a1932b4c1/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
